substratest/factory.py

```diff
@@ -196,126 +191,113 @@ def read_description(self):
 
 
-@dataclasses.dataclass
 class AlgoSpec(_AlgoSpec):
     pass
 
 
-@dataclasses.dataclass
 class AggregateAlgoSpec(_AlgoSpec):
     pass
 
 
-@dataclasses.dataclass
 class CompositeAlgoSpec(_AlgoSpec):
     pass
 
 
-@dataclasses.dataclass
 class TraintupleSpec(_Spec):
     algo_key: str
     data_manager_key: str
-    train_data_sample_keys: str
-    in_models_keys: typing.List[str]
-    tag: str
-    compute_plan_id: str
+    train_data_sample_keys: typing.List[str]
+    in_models_keys: typing.List[str] = None
+    tag: str = None
+    compute_plan_id: str = None
     rank: int = None
 
 
-@dataclasses.dataclass
 class AggregatetupleSpec(_Spec):
     algo_key: str
     worker: str
     in_models_keys: typing.List[str]
-    tag: str
-    compute_plan_id: str
+    tag: str = None
+    compute_plan_id: str = None
     rank: int = None
 
 
-@dataclasses.dataclass
 class CompositeTraintupleSpec(_Spec):
     algo_key: str
     data_manager_key: str
-    train_data_sample_keys: str
-    in_head_model_key: str
-    in_trunk_model_key: str
-    tag: str
-    compute_plan_id: str
-    out_trunk_model_permissions: typing.Dict
+    train_data_sample_keys: typing.List[str]
+    in_head_model_key: str = None
+    in_trunk_model_key: str = None
+    tag: str = None
+    compute_plan_id: str = None
+    out_trunk_model_permissions: Permissions
     rank: int = None
 
 
-@dataclasses.dataclass
 class TesttupleSpec(_Spec):
     objective_key: str
     traintuple_key: str
-    tag: str
+    tag: str = None
 
 
-@dataclasses.dataclass
-class ComputePlanTraintupleSpec:
+class ComputePlanTraintupleSpec(pydantic.BaseModel):
     algo_key: str
     data_manager_key: str
-    train_data_sample_keys: str
+    train_data_sample_keys: typing.List[str]
     traintuple_id: str
-    in_models_ids: typing.List[str]
-    tag: str
+    in_models_ids: typing.List[str] = None
+    tag: str = None
 
     @property
     def id(self):
         return self.traintuple_id
 
 
-@dataclasses.dataclass
 class ComputePlanAggregatetupleSpec(_Spec):
     aggregatetuple_id: str
     algo_key: str
     worker: str
-    in_models_ids: typing.List[str]
-    tag: str
+    in_models_ids: typing.List[str] = None
+    tag: str = None
 
     @property
     def id(self):
         return self.aggregatetuple_id
 
 
-@dataclasses.dataclass
 class ComputePlanCompositeTraintupleSpec(_Spec):
     composite_traintuple_id: str
     algo_key: str
     data_manager_key: str
-    train_data_sample_keys: str
-    in_head_model_id: str
-    in_trunk_model_id: str
-    tag: str
-    out_trunk_model_permissions: typing.Dict
+    train_data_sample_keys: typing.List[str]
+    in_head_model_id: str = None
+    in_trunk_model_id: str = None
+    tag: str = None
+    out_trunk_model_permissions: Permissions
 
     @property
     def id(self):
         return self.composite_traintuple_id
 
 
-@dataclasses.dataclass
-class ComputePlanTesttupleSpec:
+class ComputePlanTesttupleSpec(pydantic.BaseModel):
```

Comment:
```suggestion
class ComputePlanTesttupleSpec(_Spec):
```

To be consistent with other classes.